src/Sender.h

```diff
@@ -131,13 +131,14 @@ class Sender {
         /**
          * Construct an Message.
          */
-        explicit Message(Sender* sender, Driver* driver)
+        explicit Message(Sender* sender, uint16_t sourcePort)
             : sender(sender)
-            , driver(driver)
+            , driver(sender->driver)
             , TRANSPORT_HEADER_LENGTH(sizeof(Protocol::Packet::DataHeader))
             , PACKET_DATA_LENGTH(driver->getMaxPayloadSize() -
                                  TRANSPORT_HEADER_LENGTH)
             , id(0, 0)
+            , source{driver->getLocalAddress(), sourcePort}
```

Comment:
Why is the `sourcePort` not  a property of the driver like the address? Shouldn't the driver know what port it is sending on rather than the application?

Reply:
`port` is a property of message, which is transparent to lower level softwares such as transport and drivers. It's used to route messages to the right UNIX socket opened by an application. Let's discuss.